### ISSUE TITLE
Gradient end color becomes transparent

### ISSUE THREAD

**Issue by shamokit** [unocss/unocss#2522]
### UnoCSS version

0.51.5

### Describe the bug

When using a gradient class for the class attribute, the color at the end of the gradient becomes transparent.

Since the second argument of `colorToString` in the `bgGradientToValue` function is 0, `to-{toColor}` of `bg-gradient-to-r from-{startColor} via-{viaColor} to-{toColor}` will be displayed transparently.

## NG(with class attributes)
https://uno.antfu.me/play/?html=DwEwlgbgBAFgtAMwK4BsVQC4FMAeG4DGWAdtgE5QIq5QDOW1B%2BxA9sVlAIZoBcGZnYrTAYwbOABYADFIB8AKChRQkKAFtOCpUpXQqNAFZJaohAE9CJclu07w0AO5wATFNgu3AIwDmcbwPArOAwWOAoEMhY1OFoAawsIqKgIME4Y%2BLgUzkxQuIsQ2WAAensbW10oAhROWloAXgAiJ1d3Fp8-ALAgkLDKSOi8xH7k1PSLLJyx4JYGwpLIMuV5iBti0vk1haA&config=JYWwDg9gTgLgBAbwFBzgEwKYDNgDsMDCEuOA5gDQpxhQYDOGMAgjDFMAEYCuMwWAnpVQ16jAJIBjYnSHVaDGAFVcESgF84WKBBBwA5FxUS6dPUiQYAHpFjpsAQy4AbeJhz4iJYKQAUyVHQAFtAwEjx0AFyIVKh6YXQwOgC0QSFhMHpRejBWMElOpHA5lnnQ9rikGHDBAG4YUBHFeTn2TmaoarIiCpFwANoxcqJKKj4AlLLC8owsbJw8fPzjsgC6sjCBGCAYUf6ocFJO0L3IAJDnp3QA1vy7SBfnWjpZAMQALBj2EhIYepQPpxqwHsrwA7FgOABmNAADj%2B9weiVeHBRoLQACYzA9OoM1FQ8WoxkggA&css=Q&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwF8g

## OK(no class attributes)
https://uno.antfu.me/play/?html=DwEwlgbgBAFgtAMwK4BsVQC4FMAeG4DGWAdtgE5QIq5QDOW1B%2BxA9sVlAIZoBcGZnYrTAYwbOABYADFIB8AKChRQkKAFtOCpUpXQqNAFZJaohAE9CJclu07w0AO5wATFNgu3AIwDmcbwPArOAwWOAoEMhY1OFoAawsIqKgIME4Y%2BLgUzkxQuIsQ2WAAensbW10oJ1d3ap8-ALAgkLDKSOi8xDbk1PSLLJze4JZCksgy5VGIG2LS%2BRmxoA&config=JYWwDg9gTgLgBAbwFBzgEwKYDNgDsMDCEuOA5gDQpxhQYDOGMAgjDFMAEYCuMwWAnpVQ16jAJIBjYnSHVaDGAFVcESgF84WKBBBwA5FxUS6dPUiQYAHpFjpsAQy4AbeJhz4iJYKQAUyVHQAFtAwEjx0AFyIVKh6YXQwOgC0QSFhMHpRejBWMElOpHA5lnnQ9rikGHDBAG4YUBHFeTn2TmaoarIiCpFwANoxcqJKKj4AlLLC8owsbJw8fPzjsgC6sjCBGCAYUf6ocFJO0L3IAJDnp3QA1vy7SBfnWjpZAMQALBj2EhIYepQPpxqwHsrwA7FgOABmNAADj%2B9weiVeHBRoLQACYzA9OoM1FQ8WoxkggA&css=Q&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwF8g

### System Info

- macOS 13.3.1（22E261）
- Google Chrome 112.0.5615.121

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't [already an issue](https://github.com/unocss/unocss/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/unocss/unocss/discussions) or join our [Discord Chat Server](https://chat.antfu.me).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by antfu** [comment]
Would you mind sending a PR to fix it? Thanks!

**Comment by antfu** [comment]
Yeah I think we should separate the rules of the order matters to them.

**Comment by shamokit** [comment]
This is my first time making a pull request, so I apologize if there are any mistakes.

Please check.
[https://github.com/unocss/unocss/pull/2525](https://github.com/unocss/unocss/pull/2525)
